lib/rubocop/cop/variable_force/variable.rb

```diff
@@ -31,11 +33,21 @@ def initialize(name, declaration_node, scope)
         def assign(node)
           assignment = Assignment.new(node, self)
 
-          @assignments.last&.reassigned! unless captured_by_block?
+          mark_last_as_reassigned!(assignment)
 
           @assignments << assignment
         end
 
+        def mark_last_as_reassigned!(assignment)
+          return if captured_by_block?
+          return if candidate_condition?(assignment.node.parent)
```

Comment:
@koic Can I make `parent` an attribute in `RuboCop::Cop::VariableForce::Assignment`? 
Seems like a worthy refactor, makes sense as an attribute and simplifies a bunch of accesses throughout the class